pkg/controller/machine/controller.go

```diff
@@ -312,25 +312,9 @@ func (r *ReconcileMachine) getCluster(ctx context.Context, machine *machinev1.Ma
 	return cluster, nil
 }
 
-func (r *ReconcileMachine) isDeleteAllowed(machine *machinev1.Machine) bool {
-	if r.nodeName == "" || machine.Status.NodeRef == nil {
```

Comment:
Why we are removing this logic which ensures that the node on which this controller is running, will not be deleted?

Please update the PR description as well to reflect this change. Currently description sems to suggest that additionally annotation ios going to be supported.

Reply:
I can definitely imagine scenarios where the MAO node would need to be deleted, however machine-healthcheck has a special case for masters that prevents it from asking for their deletion.